website/src/containers/CodeEditorContainer.js

```diff
@@ -7,7 +7,7 @@ function mapStateToProps(state) {
   return {
     keyMap: getKeyMap(state),
     value: getCode(state),
-    mode: getParser(state).category.id,
+    mode: getParser(state).category.editorMode || getParser(state).category.id,
```

Comment:
There's no syntax highlight for Reason in codemirror... but `rust` gets us 90% of the way. In order to keep the `reason` id but be able to pass a different mode to codemirror, `editorMode` was added to category.

Reply:
@jchavarri see https://github.com/Sketch-sh/sketch-sh/blob/master/client/src_editor/reason-mode.js for hljs

Reply:
@thangngoc89 Thanks! I added this in https://github.com/fkling/astexplorer/pull/400/commits/88e9675a15a2627b8dbe73df6cfb7f76bc29a07b.

As an aside, do you know what it'd require to make this a proper codemirror mode package?

Reply:
@jchavarri I have no ideas on how to make it a package. I asked codemirror's author for upstreaming that but they said they aren't accepting any new mode. If you are willing to maintain a npm package for it, please do